proptest/src/collection.rs

```diff
@@ -597,90 +600,40 @@ impl<T: ValueTree> ValueTree for VecValueTree<T> {
-        // list until we can do so no further, then to shrink each remaining
-        // element in sequence.
-        //
-        // For `complicate()`, we simply undo the last shrink operation, if
-        // there was any.
-        if let Shrink::DeleteElement(ix) = self.shrink {
-            // Can't delete an element if beyond the end of the vec or if it
-            // would put us under the minimum length.
-            if ix >= self.elements.len()
-                || self.included_elements.count() == self.min_size
-            {
-                self.shrink = Shrink::ShrinkElement(0);
-            } else {
-                self.included_elements.clear(ix);
-                self.prev_shrink = Some(self.shrink);
-                self.shrink = Shrink::DeleteElement(ix + 1);
-                return true;
-            }
-        }
-
-        while let Shrink::ShrinkElement(ix) = self.shrink {
-            if ix >= self.elements.len() {
-                // Nothing more we can do
-                return false;
-            }
-
-            if !self.included_elements.test(ix) {
-                // No use shrinking something we're not including.
-                self.shrink = Shrink::ShrinkElement(ix + 1);
-                continue;
-            }
-
-            if !self.elements[ix].simplify() {
-                // Move on to the next element
-                self.shrink = Shrink::ShrinkElement(ix + 1);
-            } else {
-                self.prev_shrink = Some(self.shrink);
-                return true;
-            }
-        }
-
-        panic!("Unexpected shrink state");
+        false
     }
 
     fn complicate(&mut self) -> bool {
-        match self.prev_shrink {
-            None => false,
-            Some(Shrink::DeleteElement(ix)) => {
-                // Undo the last item we deleted. Can't complicate any further,
-                // so unset prev_shrink.
-                self.included_elements.set(ix);
-                self.prev_shrink = None;
-                true
-            }
-            Some(Shrink::ShrinkElement(ix)) => {
-                if self.elements[ix].complicate() {
-                    // Don't unset prev_shrink; we may be able to complicate
-                    // again.
-                    true
-                } else {
-                    // Can't complicate the last element any further.
-                    self.prev_shrink = None;
-                    false
-                }
-            }
-        }
+        false
     }
 }
 
 //==============================================================================
 // Tests
 //==============================================================================
 
-#[cfg(all(test, not(kani)))]
+#[cfg(kani)]
+mod kani_tests {
+    use super::*;
+
+    crate::proptest! {
+    #[kani::unwind(5)]
+    #[kani::proof]
+    fn vector_even_sums(
+        vec_even in vec((0..10).prop_map(|x: i32| x << 1), 0..2),
```

Comment:
`x * 2` would be clearer here (and below), but I'm not sure if you need `<<` for performance reasons.

Reply:
I don't see any difference. I will change to that.

Reply:
c5b8571ba0e8c9f98106c1a281c8bd0c076f9975